lib/SelectedObjectsProvider/useObjectSelector.js

```diff
@@ -0,0 +1,49 @@
+import { useContext } from 'react';
+import { SelectedObjectsContext } from './index';
+
+export function useObjectSelector(
+  id,
+  type,
+  idsToSelect,
+  disabledCallback = () => false
+) {
+  const {
+    selected,
+    selectMultiple,
+    deselectMultiple,
+    updateSelected,
+    currentSelectedType,
+    intendedToSelect,
+    setIntendedToSelect
+  } = useContext(SelectedObjectsContext);
+
+  const isSelected = selected.indexOf(id) !== -1;
+  const isDisabled = disabledCallback(currentSelectedType, isSelected);
+  const isHovered = intendedToSelect.indexOf(id) !== -1;
+
+  const getIds = (additionalIds = []) => [
```

Comment:
I'd be tempted to get rid of this arg if it's not used, just to reduce complexity.